language/move-lang/src/hlir/translate.rs

```diff
@@ -1378,18 +1377,39 @@ fn check_unused_locals(
     locals: &mut UniqueMap<Var, H::SingleType>,
     used: BTreeSet<Var>,
 ) -> BTreeSet<Var> {
+    let signature = context
+        .signature
+        .as_ref()
+        .expect("Signature should always be defined when checking  a function body");
     let mut unused = BTreeSet::new();
-    for (v, _) in locals.iter().filter(|(v, _)| !used.contains(v)) {
+    let mut errors = Vec::new();
+    // report unused locals
+    for (v, _) in locals
+        .iter()
+        .filter(|(v, _)| !used.contains(v) && !v.starts_with_underscore())
+    {
         let vstr = match display_var(v.value()) {
             DisplayVar::Tmp => panic!("ICE unused tmp"),
             DisplayVar::Orig(vstr) => vstr,
         };
-        let msg = format!(
-            "Unused local '{0}'. Consider removing or prefixing with an underscore: '_{0}'",
-            vstr
-        );
-        context.error(vec![(v.loc(), msg)]);
-        unused.insert(v);
+        let loc = v.loc();
+        let msg = if signature.is_parameter(&v) {
+            format!(
+                "Unused parameter '{0}'. Consider removing or prefixing with an underscore: '_{0}'",
```

Comment:
It looks close so double check this is under 120. rustfmt quietly fails on strings that it cannot format onto two lines

Reply:
Whoa, didn't know that. Unfortunate!